spark-operator/src/main/java/org/apache/spark/k8s/operator/config/SparkOperatorConf.java

```diff
@@ -0,0 +1,463 @@
+                  + "spark.kubernetes.operator.namespace, and labeled with operator name.")
+          .typeParameterClass(Boolean.class)
+          .defaultValue(false)
+          .build();
+
+  public static final ConfigOption<String> DYNAMIC_CONFIG_SELECTOR =
+      ConfigOption.<String>builder()
+          .key("spark.kubernetes.operator.dynamicConfig.selector")
+          .enableDynamicOverride(false)
+          .description("The selector str applied to dynamic config map.")
+          .typeParameterClass(String.class)
+          .defaultValue(Utils.labelsAsStr(Utils.defaultOperatorConfigLabels()))
+          .build();
+
+  public static final ConfigOption<Integer> DYNAMIC_CONFIG_RECONCILER_PARALLELISM =
+      ConfigOption.<Integer>builder()
+          .key("spark.kubernetes.operator.dynamicConfig.reconcilerParallelism")
+          .enableDynamicOverride(false)
+          .description(
+              "Parallelism for dynamic config reconciler. Unbounded pool would be used "
+                  + "if set to non-positive number.")
+          .typeParameterClass(Integer.class)
+          .defaultValue(1)
+          .build();
+
+  public static final ConfigOption<Integer> RATE_LIMITER_REFRESH_PERIOD_SECONDS =
+      ConfigOption.<Integer>builder()
+          .key("spark.kubernetes.operator.rateLimiter.refreshPeriodSeconds")
+          .enableDynamicOverride(false)
+          .description("Operator rate limiter refresh period(in seconds) for each resource.")
+          .typeParameterClass(Integer.class)
+          .defaultValue(15)
+          .build();
+
+  public static final ConfigOption<Integer> RATE_LIMITER_LIMIT_FOR_PERIOD =
+      ConfigOption.<Integer>builder()
+          .key("spark.kubernetes.operator.rateLimiter.limitForPeriod")
+          .enableDynamicOverride(false)
+          .description(
+              "Max number of reconcile loops triggered within the rate limiter refresh "
+                  + "period for each resource. Setting the limit <= 0 disables the "
+                  + "limiter.")
+          .typeParameterClass(Integer.class)
+          .defaultValue(5)
+          .build();
+
+  public static final ConfigOption<Integer> RETRY_INITIAL_INTERVAL_SECONDS =
+      ConfigOption.<Integer>builder()
+          .key("spark.kubernetes.operator.retry.initialIntervalSeconds")
+          .enableDynamicOverride(false)
+          .description("Initial interval(in seconds) of retries on unhandled controller errors.")
+          .typeParameterClass(Integer.class)
+          .defaultValue(5)
+          .build();
+
+  public static final ConfigOption<Double> RETRY_INTERVAL_MULTIPLIER =
+      ConfigOption.<Double>builder()
+          .key("spark.kubernetes.operator.retry.intervalMultiplier")
+          .enableDynamicOverride(false)
+          .description(
+              "Interval multiplier of retries on unhandled controller errors. Setting "
+                  + "this to 1 for linear retry.")
```

Comment:
Since it's `Double`, what happens when we have `0`?

Reply:
I don't think the underlying library has a sanity check for non positive values (as other rate limiting param). For this param in particular, naughty user would get max internal taking effect when setting this. https://github.com/operator-framework/java-operator-sdk/blob/main/operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/retry/GenericRetryExecution.java#L23 . which is reasonable